pkg/controller/vitessshard/reconcile_topo.go

```diff
@@ -63,40 +63,22 @@ func (r *ReconcileVitessShard) reconcileTopology(ctx context.Context, vts *plane
 	defer ts.Close()
 	wr := wrangler.New(logutil.NewConsoleLogger(), ts.Server, nil)
 
-	// Get the shard record.
-	if shard, err := ts.GetShard(ctx, keyspaceName, vts.Spec.Name); err == nil {
-		vts.Status.HasMaster = k8s.ConditionStatus(shard.HasMaster())
-		if shard.MasterAlias != nil {
-			vts.Status.MasterAlias = topoproto.TabletAliasString(shard.MasterAlias)
-		}
-
-		// Is the shard in the serving partition for any cell or tablet type?
-		if servingCells, err := ts.GetShardServingCells(ctx, shard); err == nil {
-			vts.Status.Idle = k8s.ConditionStatus(len(servingCells) == 0)
+	if *vts.Spec.TopologyReconciliation.PruneShardCells {
+		result, err := r.pruneShardCells(ctx, vts, keyspaceName, ts, wr)
+		resultBuilder.Merge(result, err)
+	}
 
-			// Clean up cells from the shard record that we don't deploy to anymore.
-			for _, cellName := range servingCells {
-				if !topo.InCellList(cellName, vts.Status.Cells) {
-					// The cell is listed in topo, but we don't deploy there anymore.
-					// We use the Vitess wrangler (multi-step command executor) to remove the cell from that shard.
-					// This is equivalent to `vtctl RemoveShardCell`.
-					if err := wr.RemoveShardCell(ctx, keyspaceName, vts.Spec.Name, cellName, false /* force*/, false /* recursive */); err != nil {
-						r.recorder.Eventf(vts, corev1.EventTypeWarning, "TopoCleanupFailed", "unable to remove cell %s from shard: %v", cellName, err)
-						resultBuilder.RequeueAfter(topoRequeueDelay)
-					} else {
-						r.recorder.Eventf(vts, corev1.EventTypeNormal, "TopoCleanup", "removed unwanted cell %s from shard", cellName)
-					}
-				}
-			}
-		} else {
-			r.recorder.Eventf(vts, corev1.EventTypeWarning, "TopoGetFailed", "failed to get shard serving cells: %v", err)
-			resultBuilder.RequeueAfter(topoRequeueDelay)
-		}
-	} else {
-		r.recorder.Eventf(vts, corev1.EventTypeWarning, "TopoGetFailed", "failed to get shard info: %v", err)
-		resultBuilder.RequeueAfter(topoRequeueDelay)
+	if *vts.Spec.TopologyReconciliation.PruneTablets {
```

Comment:
Same here. We need to still update tablet status even if we aren't pruning.

Reply:
Done. PTAL.